api/src/services/channel-service.ts

```diff
@@ -54,20 +56,33 @@ export class ChannelService {
 		return res;
 	};
 
-	getLogs = async (channelAddress: string, userId: string) => {
+	fetchLogsFromTangle = async (channelAddress: string, userId: string): Promise<ChannelData[]> => {
 		const subscription = await this.subscriptionService.getSubscription(channelAddress, userId);
 		const isAuth = subscription.type === SubscriptionType.Author;
 		const sub = await this.subscriptionPool.get(channelAddress, userId, isAuth, this.password);
 		if (!sub) {
 			throw new Error(`no author/subscriber found with channelAddress: ${channelAddress} and userId: ${userId}`);
 		}
 		const logs = await this.streamsService.getLogs(sub);
+		if (!logs) {
+			return [];
+		}
 		await this.subscriptionService.updateSubscriptionState(
 			channelAddress,
 			userId,
 			this.streamsService.exportSubscription(logs.subscription, this.password)
 		);
-		return logs;
+
+		// store logs in database
+		if (logs.channelData?.length > 0) {
+			await ChannelDataDb.addChannelData(channelAddress, userId, logs.channelData);
+		}
+		return logs.channelData;
+	};
+
+	getLogs = async (channelAddress: string, userId: string, options?: { limit: number; index: number }) => {
```

Comment:
Return value here will be something defined as return value of [this function](https://github.com/iotaledger/e-commerce-tools/pull/107/files#diff-c1294f43a2859919da2c87146ee3600698d535915ac9ec0571b16983342a6827R8)